# Declaration of Originality

**School of Informatics & IT**
<br/>**Diploma in Applied Artificial Intelligence**
<br/>**Machine Learning for Developers (CAI2C08)**
<br/>**AY2026/2027 April Semester**
<br/>**Program Codes**

* Student Name: Zhu XueLiang Anson (2500349B)



**Declaration of Originality**
* I am the originator of this work, and I have appropriately acknowledged all other original sources used as my references for this work.
* I understand that Plagiarism is the act of taking and using the whole or any part of another person’s work, including work generated by AI, and presenting it as my own.
* I understand that Plagiarism is an academic offence and if I am found to have committed or abetted the offence of plagiarism in relation to this submitted work, disciplinary action will be enforced.

# Libraries

In [1]:
## Import libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import plotly

# 1. Business Understanding
#### Goal: The purpose of this project is to develop a machine learning model to predict whether a patient is at risk of getting a heart attack 
#### based on their demographic information, medical history, lifestyle habits and clinical measurements.
#### I chose this topic as heart diseases are one of the leading causes of death worldwide. 
#### So, screening and accurately identifying patients with higher risk of heart diseases can help them receive treatment promptly before it is too late.

# 2. Data Understanding

#### The dataset is a heart disease data compiled from UCI
#### Heart disease data of patients from Cleveland, Hungary, Switzerland, and the VA Long Beach
#### The dataset has 10 features (Age, Sex, ChestPainType, RestingBP, Cholesterol, FastingBS, RestingECG, MaxHR, ExerciseAngina, OldPeak)
#### The target of the dataset is HeartDisease where 1 means yes, the person has heart diseases, and 0 means no, the person does not have heart diseases.

## 2.1 Load dataset

In [2]:
## Read *.csv file into pandas DataFrames
FILE_PATH = "mldp_project_dataset.csv"
df = pd.read_csv(FILE_PATH)

df

,Age,Sex,ChestPainType,RestingBP,Cholesterol,FastingBS,RestingECG,MaxHR,ExerciseAngina,Oldpeak,HeartDisease
0,40,1,1,140,289,0,0,172,0,0.0,0
1,49,0,2,160,180,0,0,156,0,1.0,1
2,37,1,1,130,283,0,1,98,0,0.0,0
3,48,0,3,138,214,0,0,108,1,1.5,1
4,54,1,2,150,195,0,0,122,0,0.0,0
...,...,...,...,...,...,...,...,...,...,...,...
2938,59,1,1,140,221,0,1,164,1,0.0,1
2939,60,1,0,125,258,0,0,141,1,2.8,0
2940,47,1,0,110,275,0,0,118,1,1.0,0
2941,50,0,0,110,254,0,0,159,0,0.0,1


#### there are 2943 values

## 2.2 Summary Statistics

In [4]:
## Understand the type of variable for each column
df.dtypes

Age                 int64
Sex                 int64
ChestPainType       int64
RestingBP           int64
Cholesterol         int64
FastingBS           int64
RestingECG          int64
MaxHR               int64
ExerciseAngina      int64
Oldpeak           float64
HeartDisease        int64
dtype: object

##### important to find out the datatypes 
##### if there are non-integers, one-hot-encoding would need to be performed to allow for the training to happen

In [5]:
## Check for missing data
df.isna().sum() ## Check for missing values

Age               0
Sex               0
ChestPainType     0
RestingBP         0
Cholesterol       0
FastingBS         0
RestingECG        0
MaxHR             0
ExerciseAngina    0
Oldpeak           0
HeartDisease      0
dtype: int64

##### no fields have missing value

In [6]:
## Describe data distribution
df.describe(include='all')

,Age,Sex,ChestPainType,RestingBP,Cholesterol,FastingBS,RestingECG,MaxHR,ExerciseAngina,Oldpeak,HeartDisease
count,2943.000000,2943.000000,2943.000000,2943.000000,2943.000000,2943.000000,2943.000000,2943.000000,2943.000000,2943.000000,2943.000000
mean,52.381923,0.748556,1.363575,138.698267,253.515121,0.225280,0.626911,144.040095,0.412504,1.570031,0.548420
std,13.037189,0.433917,1.142614,24.623198,112.724716,0.417838,0.713650,28.446327,0.492369,1.585102,0.497735
min,20.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,60.000000,0.000000,-2.600000,0.000000
25%,44.000000,0.000000,0.000000,120.500000,204.000000,0.000000,0.000000,124.000000,0.000000,0.100000,0.000000
50%,54.000000,1.000000,1.000000,134.000000,248.000000,0.000000,0.000000,145.000000,0.000000,1.200000,1.000000
75%,61.000000,1.000000,2.000000,150.000000,306.000000,0.000000,1.000000,165.000000,1.000000,2.400000,1.000000
max,80.000000,1.000000,3.000000,200.000000,603.000000,1.000000,2.000000,202.000000,1.000000,6.200000,1.000000


### Resting BP has min value of 0... something wrong...
### cholesterol has min value of 0... something wrong...

### rest of variables looks correct

In [7]:
print("RestingBP = 0:", (df["RestingBP"] == 0).sum())
print("Cholesterol = 0:", (df["Cholesterol"] == 0).sum())

RestingBP = 0: 1
Cholesterol = 0: 225


##### find out how many patients have restingbp, cholesterol values of 0 
##### indeed there is a patient that have restingbp as 0 which not possible
##### wow 225 patients with 0 cholesterol 
##### *FIX THOSE WITH MEDIAN* fixed with median to not get influenced by outliers

## fixing the features with median

In [8]:
#replacing the restingbp and cholesterol that has value of 0 with the median if their respective variables
df["RestingBP"] = df["RestingBP"].replace(0, df["RestingBP"].median())
df["Cholesterol"] = df["Cholesterol"].replace(0, df["Cholesterol"].median())

In [9]:
df.describe(include='all')

,Age,Sex,ChestPainType,RestingBP,Cholesterol,FastingBS,RestingECG,MaxHR,ExerciseAngina,Oldpeak,HeartDisease
count,2943.000000,2943.000000,2943.000000,2943.000000,2943.000000,2943.000000,2943.000000,2943.000000,2943.000000,2943.000000,2943.000000
mean,52.381923,0.748556,1.363575,138.743799,272.475365,0.225280,0.626911,144.040095,0.412504,1.570031,0.548420
std,13.037189,0.433917,1.142614,24.490171,86.222401,0.417838,0.713650,28.446327,0.492369,1.585102,0.497735
min,20.000000,0.000000,0.000000,80.000000,85.000000,0.000000,0.000000,60.000000,0.000000,-2.600000,0.000000
25%,44.000000,0.000000,0.000000,121.000000,219.000000,0.000000,0.000000,124.000000,0.000000,0.100000,0.000000
50%,54.000000,1.000000,1.000000,134.000000,248.000000,0.000000,0.000000,145.000000,0.000000,1.200000,1.000000
75%,61.000000,1.000000,2.000000,150.000000,306.000000,0.000000,1.000000,165.000000,1.000000,2.400000,1.000000
max,80.000000,1.000000,3.000000,200.000000,603.000000,1.000000,2.000000,202.000000,1.000000,6.200000,1.000000


In [10]:
print("RestingBP = 0:", (df["RestingBP"] == 0).sum())
print("Cholesterol = 0:", (df["Cholesterol"] == 0).sum())

#double check

RestingBP = 0: 0
Cholesterol = 0: 0


##### min for restingbp and cholesterol is not 0 anymore
##### the min for restingbp is 80
##### the min for cholesterol is 85

## 2.3 Data Visualization

### 2.3.1 Understanding distribution of data

### 2.3.1.1 Understanding distribution of target

In [ ]:
## Understanding distribution of target


### 2.3.1.2 Understanding distribution of features

In [ ]:
## Understanding distribution of features


### 2.3.2 Understanding relationship between variables

In [ ]:
## Understanding relationship between variables


# 3. Data Preparation

## 3.1 Data Cleaning

In [ ]:
## Clean data

## 3.2 Train-Test Split

In [ ]:
## Split data into train set and test set


# 4. Modelling

### 4.2 Train Model

In [ ]:
## Initialise and train model


# 5. Model Evaluation

In [ ]:
## Evaluate model


In [ ]:
## New data

## Predict


## Iterative model development


In [ ]:
## Further feature engineering / feature selection